cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -144,7 +151,12 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 			continue
 		}
 
+		nodeGroupsWithExpansionOption[nodeGroup.Id()] = true
 		options = append(options, option)
+
+		if useBinpackingLimiter && o.processors.BinpackingLimiter.StopBinpacking(o.autoscalingContext, options, nodeGroupsWithExpansionOption) {
```

Comment:
I don't get it.  The idea of `StopBinpacking` method is to stop the binpacking early by looking at the options (even if they are empty) and processNodeGroups. It's the method definition which would return bool to take a decision.

Reply:
I don't think it would help anyway. If we do not have an option and call `StopBinpacking`, we are basically checking at the same state as in the previous iteration. I mean the state of the method is not changing, `StopBinpacking` takes (autoscalingContext, options, processedNodeGroups) which is not changed if we have no options. Do you think we should have more info in `processedNodeGroups`?
Currently `processedNodeGroups[nodegroup_id] = true` means node is processed (either it has an expansion option or it is skipped) and `false` means either it is unprocessed or without an expansion option.

Reply:
Or `processedNodeGroups` just means that we have processed node groups uptil here irrespective of the options they have or not?